### DISCUSSION TITLE
GPU benchmarks

### DISCUSSION THREAD

**Discussion by akcopy** [milvus-io/milvus#30200] [Q&A and General discussion] [answered]
I'm looking to deploy a Milvus instance, however I'm not sure which GPUs would perform best in terms of cost effectiveness for Milvus. Are there any internal benchmarks to see what Milvus performance on GPU depends on the most? For eg CUDA cores or memory size or memory speed?

**Comment by yhmo** [comment]
You can use the official benchmark tool to test: https://github.com/zilliztech/VectorDBBench

We don't have a detailed benchmark for different cores, memory size, or memory speed.

**Reply by akcopy** [reply]
has there been any internal testing which I could glean some insights from? For e.g. i'm thinking about 2x rtx3090 vs 1x rtx4090 - which would be more effective?

**Reply by yhmo** [reply]
We tested different graphic cards, including 3090. With 1 million rows dataset. The QPS of GPU_IVF_FLAT index is 5X ~ 6X faster than CPU index(HNSW).

**Reply by akcopy** [reply]
thanks! Do you know how the different graphic cards compare with each other in terms of speed (say between Tesla P100, RTX 3090, RTX 4090)?

**Reply by yhmo** [reply]
We don't know.
I think the performance is no much difference between 3090 and 4090.

**Reply by akcopy** [reply]
why do you think so? I would have thought that 4090 would be much faster than 3090. Is it because it would be bottlenecked by the memory? (both have the same RAM capacity)

**Reply by yhmo** [reply]
I don't have 4090, ignore my opinion. 4090 is a great graphic card for 3A games.

**Comment by xiaofan-luan** [comment, marked as answer]
We tried 3090, T4 and L40.
What we recommend is to use card from the above list. Among those T4 might be easiest to get
